tachyon/zk/expressions/evaluator/selector_replacer.h

```diff
@@ -18,9 +18,8 @@ namespace tachyon::zk {
 template <typename F>
 class SelectorsReplacer : public Evaluator<F, std::unique_ptr<Expression<F>>> {
  public:
-  SelectorsReplacer(
-      const std::vector<std::unique_ptr<Expression<F>>>& replacements,
-      bool must_be_non_simple)
+  SelectorsReplacer(const std::vector<const Expression<F>*>& replacements,
+                    bool must_be_non_simple)
```

Comment:
how about using `Ref<Expression<F>>` instead of `Expression<F>*`?
(defined in tachyon/zk/base/ref.h)